pkg/api/rest/resttest/resttest.go

```diff
@@ -557,6 +580,40 @@ func (t *Tester) testDeleteNonExist(obj runtime.Object) {
 
 }
 
+func (t *Tester) testDeleteWithUID(obj runtime.Object, setFn SetFunc, getFn GetFunc, isNotFoundFn IsErrorFunc) {
+	ctx := t.TestContext()
+
+	foo := copyOrDie(obj)
+	t.setObjectMeta(foo, "foo1")
+	objectMeta := t.getObjectMetaOrFail(foo)
+	objectMeta.UID = types.UID("UID0000")
+	if err := setFn(ctx, foo); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	obj, err := t.storage.(rest.GracefulDeleter).Delete(ctx, objectMeta.Name, api.NewUIDDeleteOptions("UID1111"))
```

Comment:
This tests the fast-fail path; the etcd path is tested elsewhere, correct? Can you add a comment to this effect?